### DISCUSSION TITLE
ConstructorHelpers::FObjectFinder<T>함수와 StaticLoadObject 함수의 차이점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by simeddk** [simeddk/Tore04_BasicSyntaxCPP#2] [Q&A] [answered]
2개의 함수는 프로젝트 콘텐츠 내부의 에셋을 C++ 코드로 소환시키는 함수입니다.
그런데 에픽게임즈는 할 일이 없나요? 왜 같은 기능의 함수를 두개나 만든거죠?(~~사실 찾아보면 더 있다 우하하~~)
혹시 저를 괴롭히려고 일부러 그런건 아니겠죠?

**Comment by HyeonLang** [comment, marked as answer]
김재우
ConstructorHelpers::FObjectFinder<T>는 생성자에서만 호출될 수 있습니다.
런타임에 에셋을 호출하려면 StaticLoadObject을 사용해야합니다.

**Comment by ehddnjs1200** [comment]
정동원

에픽게임즈는 같은 기능을 하는 함수가 여러 개 있는 것은 개발자 경험을 향상시키기 위한 의도된 설계입니다.
StaticLoadObject는 함수는 특정 경로에 있는 에셋을 로드하는 데 사용됩니다. 주로 런타임에 동적으로 에셋을 로드할 때 사용되고
ConstructorHelpers::FObjectFinder는 생성자에서 에셋을 찾는 데 사용되며, 주로 에디터에서 미리 로드된 에셋을 참조할 때 사용됩니다
이처럼 에픽게임즈는 개발자들이 필요에 따라 가장 적합한 도구를 선택할 수 있도록 다양한 로드 함수를 제공합니다. 이는 개발자들에게 유연성과 제어 권한을 주기 위한 것이며, 다양한 상황에서 최적화된 성능을 제공할 수 있도록 돕습니다.
